### ISSUE TITLE
weight here will lead to zero loss in the dice loss

### ISSUE THREAD

**Issue by Boyden** [open-mmlab/mmdetection#6416]
https://github.com/open-mmlab/mmdetection/blob/db256a14bb7018ad36eed104ea0ce178a0d4050c/mmdet/models/losses/utils.py#L43
In the `mmdet.models.roi_heads.mask_heads.fcn_mask_head.py#L173`, the weight was initialized to zeros using `torch.zeros_like(labels)`. 

It will lead to zero loss when call `loss = weight_reduce_loss(loss, weight, reduction, avg_factor).` in the `mmdet.models.losses.dice_loss.py#L44`

I suggest that add 
```python
    if weight.sum() == 0:
        weight = torch.ones_like(weight)
    else:
        weight = weight/weight.sum()
```
before call `mmdet.models.losses.dice_loss.py#L44`

**Comment by hhaAndroid** [comment]
@Boyden Thank you for your feedback, there is indeed a potential problem. It seems more appropriate to modify `fcn_mask_head` directly, because dice loss is not responsible for this logic.  Are you interested to create PR?

**Comment by Boyden** [comment]
Directly modifying the `fcn_mask_head` will lead to inconsistency when calling other loss functions such as `CrossEntropyLoss`. The more appropriate way I think is to modify the `dice_loss` code.
